app/models/post_image_manager.rb

```diff
@@ -0,0 +1,34 @@
+require 'singleton'
+
+##
+# A singleton class for managing all image attachments for a post
+class PostImageManager
+  include Singleton
+  
+  attr_reader :uploaders
+
+  ##
+  # The constructor for PostImageManager which initializes the array of Carrierware
+  # image uploaders to use when submiting a post
+  def initialize
+    @uploaders = []
+  end
+
+  ##
+  # Adds an image to be uploaded in a SSE website post
+  #
+  # Params:
+  # +file+:: A ActionDispatch::Http::UploadedFile object containing the file to be used in a post
+  def add_file(file)
+    uploader_to_add = PostImageUploader.new
+    uploader_to_add.cache!(file)
+    @uploaders.delete_if { |x| x.filename == file.original_filename }
```

Comment:
This looks to be preventing duplicates? Is that correct?

Reply:
Yes that's correct. I added the statement above to prevent against any issues for instance if a user wants to move their image to a different location on the post. If they delete the tag the previous uploader will still be in the cache, so that's why I look for duplicate uploaders when a user drags and drops an image onto a post

Reply:
Perfect. Thanks for the explanation!